### PR TITLE
fix: don't double-prepend basehref to redirect URLs (fixes #3137)

### LINKED ISSUE #3137: ArgoCD UI: login redirect loop
Checklist:

* [ ] I've searched in the docs and FAQ for my answer: http://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

When installing ArgoCD at a base href (e.g., "/argocd") and accessing that base href, you get redirected to /argocd/applications, and then immediately redirected to /argocd/argocd/login?...

This appears to enter a redirect loop.

**To Reproduce**

1. Install ArgoCD and specify base-href=/argocd
1. Access the UI at /argocd
1. Observe redirect loop.

**Expected behavior**

It should redirect to /argocd/login 

**Version**

This is a ArgoCD v1.4.2 deployment:
```
$ cat Chart.yaml
apiVersion: v1
appVersion: "1.4.2"
description: A Helm chart for ArgoCD, a declarative, GitOps continuous delivery tool for Kubernetes.
name: argo-cd
version: 1.8.0
home: https://github.com/argoproj/argo-helm
icon: https://raw.githubusercontent.com/argoproj/argo/master/docs/assets/argo.png
keywords:
  - argoproj
  - argocd
  - gitops
maintainers:
  - name: alexec
  - name: alexmt
  - name: jessesuen
  - name: seanson
```

**Logs**

```
Paste any relevant application logs here.
```
Well, I think the issue is here:
https://github.com/argoproj/argo-cd/blob/7fde387dd631a9236876dd54cb6a371430d5fbfc/ui/src/app/app.tsx#L72

That line should probably check for `${basehref}/login` instead of `/login`.

### PR DESCRIPTION
**Pull request by gpaul**
Fixes https://github.com/argoproj/argo-cd/issues/3137

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Optional. My organization is added to the README.
* [x] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

### CODE REVIEW

**Comment by gpaul** [comment]
@alexmt mind taking a look? I can confirm this fixes the issue.

File: ui/src/app/app.tsx [hunk 1]
```diff
@@ -85,9 +85,9 @@ requests.onError.subscribe(async err => {
             .getAttribute('href')
             .replace(/\/$/, '');
         if (isSSO) {
-            window.location.href = `${basehref}/auth/login?return_url=${encodeURIComponent(location.href)}`;
```
Comment (alexmt) [inline ui/src/app/app.tsx]: It appears that this change is not required for SSO flow. Looks like `history.push` taking into account page basehref but `window.location.href` don't. so we still need to has set `basehref` manually here. Can you please revert this line?

**Review by alexmt — CHANGES_REQUESTED** [review]
Thanks you for the fix @gpaul ! Requested one minor change

**Comment by gpaul** [comment]
Ah right, done.

**Review by alexmt — APPROVED** [review]

**Pull request merged**